walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -19,24 +19,87 @@
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
-/**
- * Given a list of events, returns a list of possible time ranges for the meeting request.
- */
 public final class FindMeetingQuery {
+
+  /** Create a meeting timeslot if there is enough time between firstTime and secondTime. */
+  private Optional<TimeRange> createDuration(long duration, int firstTime, int secondTime, 
+    boolean inclusive) {
+    if ((secondTime - firstTime) >= duration) {
+      return Optional.of(TimeRange.fromStartEnd(firstTime, secondTime, inclusive));
+    }
+    return Optional.empty();
+  }
+
+  /** Returns a list of time slots that work for mandatory attendees. */
+  private List<TimeRange> findAvailabilityForMandatoryAttendees(
+    List<TimeRange> eventTimes, long duration) {
```

Comment:
For duration for capstone and future PRs starting from scratch, prefer non-primitives -- ie Duration clas (https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html).